cumulusci/tasks/bulkdata/step.py

```diff
@@ -264,6 +263,17 @@ def get_results(self):
 class BulkApiDmlOperation(BaseDmlOperation, BulkJobMixin):
     """Operation class for all DML operations run using the Bulk API."""
 
+    def __init__(self, *, sobject, operation, api_options, context, fields):
+        super().__init__(
+            sobject=sobject,
+            operation=operation,
+            api_options=api_options,
+            context=context,
+            fields=fields,
+        )
+        self.csv_buff = io.StringIO()
```

Comment:
I think we should pass `newline=""` here so that the csv writer can manage the newlines correctly if there are records with text that includes newlines.

Reply:
(But make sure that we're counting a newline at the end of each record toward the limit.)

Reply:
@davisagli - We serialize the records to `.csv` format (write with the csv writer and then read the record back from the `IO.StringIO(newline='')` stream) and only then do we count the number of bytes. It looks like specifying newline as `''` is indicating that the stream should not do any translations with newline chars. Is this also adding an additional newline somewhere in the mix?

Reply:
@Br4nd0R I'm saying we should verify that serializing a list like ["col 1", "multiline\ncol 2"] results in `col 1,"multiline\ncol 2"\n` and counts as 24 characters.